pkg/agent/ipassigner/ip_assigner_linux.go

```diff
@@ -83,9 +92,41 @@ func NewIPAssigner(nodeTransportInterface string, dummyDeviceName string) (*ipAs
 			return nil, fmt.Errorf("error when loading IP addresses from the system: %v", err)
 		}
 	}
+	// Copy the assigned IPs on the dummy interface to ARP responder or NDP responder.
+	for assigned := range a.assignedIPs {
+		ip := net.ParseIP(assigned)
+		var err error
+		if utilnet.IsIPv4(ip) && a.arpResponder != nil {
+			err = a.arpResponder.AddIP(ip)
```

Comment:
I found this will unconditionally trigger a GARP, which may disrupt existing connection if this node previously has one IP assigned and the agent restarts? Should we add a `ReplaceIPs` method which is supposed to be called on initialization and only add desired IPs to responders?

Reply:
Added ReplaceIPs interface to the responder. thanks!